applications/FluidDynamicsApplication/python_scripts/adaptative_remeshing_fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+
+# Other imports
+import sys
+
+# Import the base structural analysis
+from KratosMultiphysics.FluidDynamicsApplication.fluid_dynamics_analysis import FluidDynamicsAnalysis as BaseClass
+
+class AdaptativeRemeshingFluidDynamicsAnalysis(BaseClass):
+    """
+    This class is the main-script of the FluidDynamicsApplication when using adaptative remeshing put in a class
+
+    It can be imported and used as "black-box"
+    """
+
+    def RunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage
+        It can be overridden by derived classes
+        """
+
+        # If we remesh using a process
+        computing_model_part = self._GetSolver().GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+
+        while self.time < self.end_time:
+            self.time = self._GetSolver().AdvanceInTime(self.time)
+            # We reinitialize if remeshed previously
```

Comment:
Shouldn't the re-meshing always be done in the same moment? These if statements can be easily misunderstood...

Reply:
Nope, you may be interested in doing the remeshing during the simulation, not before, and then you need to reinitialize the whole simulation

Reply:
But this means that you might do the re-meshing before and after the `InitializeSolutionStep()`. When does this have sense?
If the re-meshing is to be done once, and we want to call it as a process, I'd rather go for a interval-type solution as we use in the processes.